### PR TITLE
Tarun samanta/#15

### PR DESCRIPTION
**Pull request by SamantaTarun**
added Bit manipulations fundamental questions with solutions

### CODE REVIEW

**Comment by tarunsamanta2k20** [comment]
please take this branch as well. I removed some comments from codes from previous commits.

**Comment by ShehrozIrfan** [comment]
Hey @tarunsamanta2k20 
Good work, I've checked the files and it looks good.

I'm just wondering are these topics are related to data structures? OR I think we can add these in the programming fundamentals repo. Here is the link to that you can visit:
[Programming Fundamentals](https://github.com/ShehrozIrfan/learn-programming-fundamentals)

**Comment by SamantaTarun** [comment]
Hey @ShehrozIrfan 

Yes this Bit-Manipulation most asked interview topic under Data
Structure and Algorithm (DSA). By reading the content, one can easily solve
questions in any interview. You must add these questions.

File: C++/13-Bit Manipulation/1 counting the number of set bits.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+	#include<bits/stdc++.h>
```
Comment (ShehrozIrfan) [inline C++/13-Bit Manipulation/1 counting the number of set bits.cpp]: Please change the file names to something like the below:
`1-file_name.cpp`
`2-file_name.cpp`
etc

File: C++/13-Bit Manipulation/4 bitwise right shift.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		while(t--)
+		{
+			int n,i;
+			
+			cin>>n>>i;
+			
+			//Right shift n by i bits i.e n=n/(2^i)
+			
+			cout<<n<<">>"<<i<<" = "<<(n>>i)<<endl;
+
```
Comment (ShehrozIrfan) [inline C++/13-Bit Manipulation/4 bitwise right shift.cpp]: Remove the extra lines.

**Review by ShehrozIrfan — CHANGES_REQUESTED** [review]
Hey @tarunsamanta2k20 

Overall looks good.

Just some minor updates:
- Need to change the file names
- Need to remove the extra lines
  - Make sure there are no extra lines between statements and remove the un-necessary extra lines from all the files.

**Comment by SamantaTarun** [comment]
Okay making changes.

**Comment by SamantaTarun** [comment]
@ShehrozIrfan don't remove the commented lines. Those lines actually describing the question and explaining the logic for the code.

File: C++/13-Bit Manipulation/11-Power_of_two.cpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
+  #include<bits/stdc++.h>
```
Comment (ShehrozIrfan) [inline C++/13-Bit Manipulation/11-Power_of_two.cpp:1]: Is this related to `Bits` or `DSA`? I guess this one should be in the programming fundamentals. What do you think?

File: C++/13-Bit Manipulation/6-is_power_of_2.cpp [hunk 4]
```diff
@@ -0,0 +1,21 @@
+		#include<bits/stdc++.h>
```
Comment (ShehrozIrfan) [inline C++/13-Bit Manipulation/6-is_power_of_2.cpp:1]: Is this related to `Bits` or `DSA`? I guess this one should be in the programming fundamentals. What do you think?

**Review by ShehrozIrfan — COMMENTED** [review]
Overall Excellent.

Just some questions, need your thoughts.

**Comment by SamantaTarun** [comment]
Hey ShehrozIrfan,

This is an header file. If you include this header file all other header
files will be imported automatically.

No problem with that. You can merge now.

File: C++/13-Bit Manipulation/6-is_power_of_2.cpp (hunk 4 above)
Comment (SamantaTarun) [reply C++/13-Bit Manipulation/6-is_power_of_2.cpp:1]: Hey @ShehrozIrfan 

No. Why are you bothering for small things. This is basically an advance header file. If you include this header file then all the header files will be imported automatically.

Comment (ShehrozIrfan) [reply C++/13-Bit Manipulation/6-is_power_of_2.cpp:1]: Hey @tarunsamanta2k20 

My apologies for not making the point clear.

Actually, I was not asking about the header file, I was asking about the file like it says the `program to check if the integer is the power of 2`, so I was wondering if we can add this to `programming fundamentals` or not.
What do you think?

Comment (SamantaTarun) [reply C++/13-Bit Manipulation/6-is_power_of_2.cpp:1]: @ShehrozIrfan 

No need to add to programming fundamental. This bit manipulation topic is under DSA then you must add to your DSA branch i.e learn-dsa. Merge it now. I will share more DSA topic in future.

**Comment by SamantaTarun** [comment]
Hey @ShehrozIrfan 

No need to add to programming fundamental. This bit manipulation topic is under DSA then you must add to your DSA branch i.e learn-dsa. Merge it now. I will share more DSA topic in future

**Review by ShehrozIrfan — APPROVED** [review]

**Comment by ShehrozIrfan** [comment]
Hey @tarunsamanta2k20 

Great work.

You can star this repo for further updates.

Have fun 🙌🏼
